core/input/input.cpp

```diff
@@ -288,10 +288,14 @@ bool Input::is_joy_button_pressed(int p_device, int p_button) const {
 }
 
 bool Input::is_action_pressed(const StringName &p_action) const {
+	bool has_action = InputMap::get_singleton()->has_action(p_action);
+	ERR_FAIL_COND_V_MSG(!has_action, false, "Request for nonexistent InputMap action '" + String(p_action) + "'.");
```

Comment:
How about wrapping those in `DEBUG_ENABLED` for performance in `release` (if that makes a difference)?

```suggestion
#ifdef DEBUG_ENABLED
	bool has_action = InputMap::get_singleton()->has_action(p_action);
	ERR_FAIL_COND_V_MSG(!has_action, false, "Request for nonexistent InputMap action '" + String(p_action) + "'.");
#endif
```

Reply:
ah, good point.
I was wondering about the performance impact too but totally forgot about this option, thanks :)